satellite/rewards/rewards.go

```diff
@@ -89,53 +92,52 @@ type Offer struct {
 	Type   OfferType
 }
 
-// IsDefault evaluates the default status of offers for templates.
-func (o Offer) IsDefault() bool {
-	if o.Status == Default {
-		return true
-	}
-	return false
-}
-
-// IsCurrent evaluates the current status of offers for templates.
-func (o Offer) IsCurrent() bool {
-	if o.Status == Active {
-		return true
-	}
-	return false
-}
+// Offers contains a slice of offers.
+type Offers []Offer
 
-// IsDone evaluates the done status of offers for templates.
-func (o Offer) IsDone() bool {
-	if o.Status == Done {
-		return true
-	}
-	return false
+// OrganizedOffers contains a list of offers organized by status.
+type OrganizedOffers struct {
+	Active  Offer
+	Default Offer
+	Done    Offers
 }
 
-// Offers holds a set of organized offers.
-type Offers struct {
-	Set []Offer
+// OfferSet provides a separation of marketing offers by type.
+type OfferSet struct {
+	ReferralOffers OrganizedOffers
+	FreeCredits    OrganizedOffers
 }
 
-// GetCurrentFromSet returns the current offer from an organized set.
-func (offers Offers) GetCurrentFromSet() Offer {
-	var o Offer
-	for _, offer := range offers.Set {
-		if offer.IsCurrent() {
-			o = offer
+// OganizeOffersByStatus organizes offers by OfferStatus.
+func (offers Offers) OganizeOffersByStatus() (oo OrganizedOffers) {
+	for _, offer := range offers {
```

Comment:
what if `offers` is empty?

Reply:
You get an empty OrganizedOffers struct. Nothing will render to the template and there will be no error thrown.